fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -348,17 +353,45 @@ public CompletableFuture<Boolean> markIndexReadable(boolean markReadablePlease)
         if (!markReadablePlease) {
             return AsyncUtil.READY_FALSE; // they didn't say please..
         }
-        AtomicBoolean allReadable = new AtomicBoolean(true);
-        return getRunner().runAsync(context -> openRecordStore(context)
-                .thenCompose(store -> forEachTargetIndex(index ->
-                        store.markIndexReadable(index).thenApply(built -> {
-                            if (!built) {
-                                allReadable.set(false);
-                            }
-                            return null;
-                        }))
-                        .thenApply(ignore -> allReadable.get())
-                ), common.indexLogMessageKeyValues("IndexingBase::markReadable"));
+
+        AtomicReference<RecordIndexUniquenessViolation> uniquenessViolationExRef = new AtomicReference<>();
+        AtomicBoolean allReadable = new AtomicBoolean();
```

Comment:
This variable might be misleading now. It should be `true` if no states had to change, correct? That would be the implication if `markReadablePlease` is false.